### PR TITLE
pkg/flare: add trace-agent expvars into flare, when available.

### PR DESCRIPTION
**Pull request by gbbr**
This change adds collecting of expvars from trace-agent into the flare,
when available.

### CODE REVIEW

File: pkg/flare/archive.go [hunk 1]
```diff
@@ -326,7 +326,32 @@ func zipExpVar(tempDir, hostname string) error {
...
 	}
 
-	return nil
+	apmPort := "8126"
+	if config.Datadog.IsSet("apm_config.receiver_port") {
+		apmPort = config.Datadog.GetString("apm_config.receiver_port")
+	}
+	resp, err := http.Get(fmt.Sprintf("http://127.0.0.1:%s/debug/vars", apmPort))
+	if err != nil || resp.StatusCode != http.StatusOK {
+		// trace-agent may not be running
```
Comment (hush-hush) [inline pkg/flare/archive.go]: We should write that in the file (with the error code and the error ?). That way when we receive a flare we will know exactly why we were not able to get the info from the trace-agent.

File: pkg/flare/archive.go [hunk 2]
```diff
@@ -326,7 +326,32 @@ func zipExpVar(tempDir, hostname string) error {
 		}
 	}
 
-	return nil
+	apmPort := "8126"
+	if config.Datadog.IsSet("apm_config.receiver_port") {
+		apmPort = config.Datadog.GetString("apm_config.receiver_port")
+	}
+	resp, err := http.Get(fmt.Sprintf("http://127.0.0.1:%s/debug/vars", apmPort))
+	if err != nil || resp.StatusCode != http.StatusOK {
```
Comment (hush-hush) [inline pkg/flare/archive.go]: When will `err != nil` but `StatusCode` be `StatusOK` ?

**Review by hush-hush — CHANGES_REQUESTED** [review]

File: pkg/flare/archive.go (hunk 2 above)
Comment (gbbr) [reply pkg/flare/archive.go]: It's an _OR_ condition. If err is not nil, it executes _OR_ if status is not OK 😄 You misread.

File: pkg/flare/archive.go (hunk 1 above)
Comment (gbbr) [reply pkg/flare/archive.go]: Good idea! Changed and wrote to file in both cases: err != nil and resp != 200

**Review by gbbr — COMMENTED** [review]
Thanks @hush-hush. Applied the changes you've suggested. PTAL

File: pkg/flare/archive.go (hunk 2 above)
Comment (hush-hush) [reply pkg/flare/archive.go]: :man_facepalming:

**Review by hush-hush — APPROVED** [review]

**Pull request merged**